lib/Doctrine/ODM/MongoDB/Aggregation/Builder.php

```diff
@@ -180,9 +188,12 @@ public function execute(array $options = []): Iterator
         return $this->getAggregation($options)->getIterator();
     }
 
+    /**
+     * @deprecated use createExpr instead
+     */
     public function expr(): Expr
     {
-        return new Expr($this->dm, $this->class);
+        return $this->createExpr();
```

Comment:
We also trigger a deprecation notice that users can react to according to their implementation. To do so, use the `trigger_deprecation` helper:
```suggestion
        trigger_deprecation(
            'doctrine/mongodb-odm',
            '2.3',
            'The "%s" method is deprecated. Please use "%s::createAggregationExpr" instead.',
            __METHOD__,
            static::class
        );
        
        return $this->createAggregationExpression();
```